charon/tests/ui/issue-120-bare-discriminant-read.out

```diff
@@ -9,21 +9,17 @@ fn test_crate::call_is_some<T>(@1: core::option::Option<T>) -> bool
 {
     let @0: bool; // return
     let opt@1: core::option::Option<T>; // arg #1
-    let self@2: &'_ (core::option::Option<T>); // local
-    let @3: isize; // anonymous local
+    let @2: isize; // anonymous local
 
-    self@2 := &opt@1
     match opt@1 {
-        0 => {
-            @3 := const (0 : isize)
-        },
         1 => {
-            @3 := const (1 : isize)
+            @0 := const (true)
+        },
+        0 => {
+            @0 := const (false)
         }
```

Comment:
I'm surprised that the order changed here (case 1 then case 0)

Reply:
I think it changed in the MIR (and this is likely my doing again).

Reply:
Oh wait, that's code that we generate ourselves. We generate it in the order of the variants in the type definition, maybe that changed? That _is_ weird.

Reply:
I opened an issue https://github.com/AeneasVerif/charon/issues/278